mlflow/models/signature.py

```diff
@@ -306,14 +306,24 @@ def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
     if hints.input is None:
         return None
 
+    params = None
+    if isinstance(input_example, dict) and "params" in input_example:
+        params = input_example.pop("params")
+        if len(input_example) == 1:
+            input_example = list(input_example.values())[0]
+
     input_schema = _infer_schema_from_type_hint(hints.input, input_example) if hints.input else None
-    output_example = func(input_example) if input_example else None
+    if params and inspect.signature(func).parameters.get("params"):
```

Comment:
```suggestion
    if params and "params" in inspect.signature(func).parameters:
```

`.get("params")` looks like we also care about the value of `parameters["params"]`, but we don't.

Reply:
Can we verify param keys validity here ?

Reply:
Moving 'params_schema = _infer_param_schema(params) if params else None' up, this validates keys as well.